cpp/src/utilities/cython.cu

```diff
@@ -526,6 +526,76 @@ void call_pagerank(raft::handle_t const& handle,
   }
 }
 
+// Wrapper for calling Katz centrality through a graph container
+template <typename vertex_t, typename weight_t>
+void call_katz_centrality(raft::handle_t const& handle,
+                          graph_container_t const& graph_container,
+                          vertex_t* identifiers,
+                          weight_t* katz_centrality,
+                          double alpha,
+                          double beta,
+                          double tolerance,
+                          int64_t max_iter,
+                          bool normalized)
+{
+  if (graph_container.graph_type == graphTypeEnum::GraphCSRViewFloat) {
```

Comment:
why not a switch statement vs lots of if-else?

Reply:
This needs to be addressed/overwritten when having a new generic type dispatching code